src/compiler/utilities.ts

```diff
@@ -4235,6 +4251,65 @@ namespace ts {
         return options.watch && options.hasOwnProperty("watch");
     }
 
+    /**
+     * Given a name and a list of names are *not* equal to the name, return a spelling suggestion if there is one that is close enough.
+     * Names less than length 3 only check for case-insensitive equality, not Levenshtein distance.
+     */
+    export function getSpellingSuggestion(name: string, choices: string[]): string | undefined;
+
+    /**
+     * Given a name and a list of names are *not* equal to the name, return a spelling suggestion if there is one that is close enough.
+     * Names less than length 3 only check for case-insensitive equality, not Levenshtein distance.
+     */
+    export function getSpellingSuggestion<T>(name: string, choices: T[], exclusions: string[] | undefined, getName: (candidate: T) => string | undefined): T | undefined;
+    export function getSpellingSuggestion<T>(name: string, choices: T[], exclusions?: string[], getName: (candidate: T) => string | undefined = identity): T | undefined {
+        // If there is a candidate that's the same except for case, return that.

```

Comment:
I didn't want to repeat the rather lengthy description for both overloads, and its a lot of text to show when you mouse over a reference. Its too bad we don't support the `@summary` jsdoc tag.